simple_salesforce/metadata.py

```diff
@@ -322,16 +321,11 @@ def _read_deploy_zip(zipfile):
         :rtype:
         """
         if hasattr(zipfile, 'read'):
-            file = zipfile
-            file.seek(0)
-            should_close = False
+            zipfile.seek(0)
+            raw = zipfile.read()
         else:
-            file = open(zipfile, 'rb')
-            should_close = True
-        raw = file.read()
-        if should_close:
-            file.close()
-        return b64encode(raw).decode("utf-8")
+            raw = Path(zipfile).read_bytes()
+        return b64encode(raw).decode()
```

Comment:
Simplified code to not need the should_close variable. Also, it's opening and closing the file in the Path.read_bytes method.

I preferred this simpler solution than:
```
        try:
            raw = file.read()
        finally:
            if should_close:
                file.close()
```
which is equivalent. (code before commit was missing the finally block)